src/components/LoginPage/LoginPage.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { Form, Field } from 'react-final-form';
+import { routes } from '../../constants';
+import { history } from '../../helpers';
+
+class LoginPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      user: {
+        email: '',
+        password: '',
+      },
+      error: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    if (localStorage.getItem('user')) {
+      history.push('/');
+    }
+  }
+
+  handleChange(e) {
+    const { name, value } = e.target;
+    const { user } = this.state;
```

Comment:
use setState callback function if you are reading the state again while setting state